src/ai/backend/manager/models/agent.py

```diff
@@ -325,6 +331,174 @@ class Meta:
     items = graphene.List(Agent, required=True)
 
 
+async def _query_domain_groups_by_ak(
+    db_conn: SAConnection,
+    access_key: str,
+) -> tuple[str, list[uuid.UUID]]:
+    j = sa.join(
+        keypairs,
+        users,
+        keypairs.c.user == users.c.uuid,
+    )
+    query = (
+        sa.select([keypairs, users.c.domain_name])
+        .select_from(j)
+        .where(keypairs.c.access_key == access_key)
+    )
+    result = await db_conn.execute(query)
+    row = result.first()
+    domain_name = row["domain_name"]
+    user_id = row["user"]
+
+    query = (
+        sa.select([association_groups_users.c.group_id])
+        .select_from(association_groups_users)
+        .where(association_groups_users.c.user_id == user_id)
+    )
+    result = await db_conn.execute(query)
+    groud_ids = [row["group_id"] for row in result.fetchall()]
+    return domain_name, groud_ids
+
+
+async def _append_sgroup_from_clause(
+    graph_ctx: GraphQueryContext,
+    query: sa.sql.Select,
+    access_key: str,
+    scaling_group: str | None,
+):
+    if scaling_group is not None:
+        query = query.where(agents.c.scaling_group == scaling_group)
+    else:
+        async with graph_ctx.db.begin_readonly() as conn:
+            domain_name, group_ids = await _query_domain_groups_by_ak(conn, access_key)
+            sgroups = await query_allowed_sgroups(conn, domain_name, group_ids, access_key)
+        query = query.where(agents.c.scaling_group.in_([sgroup["name"] for sgroup in sgroups]))
+    return query
+
+
+class AgentSummary(graphene.ObjectType):
+    """
+    A schema for normal users.
+    """
+
+    class Meta:
+        interfaces = (Item,)
+
+    status = graphene.String()
+    scaling_group = graphene.String()
```

Comment:
Is it necessary?

Reply:
https://teams.microsoft.com/l/message/19:3c0acc0825024daeb2211f55c4e7f4aa@thread.skype/1660193938601?tenantId=13c6a44d-9b52-4b9e-aa34-0513ee7131f2&groupId=74ae2c4d-ec4d-4fdf-b2c2-f5041d1e8631&parentMessageId=1660193938601&teamName=devops&channelName=Backend.AI%20Talks&createdTime=1660193938601&allowXTenantAccess=false

related link of this.
scaling_group can be `None`, then it will fetch allowed scaling groups for the user.

Reply:
I should have checked, but I couldn't. Thanks!